tests/test_docker/test_dataset.py

```diff
@@ -190,7 +190,7 @@ def test_resisc45(self):
             self.skipTest("resisc45_split dataset not locally available.")
 
         for split, size in [("train", 22500), ("validation", 4500), ("test", 4500)]:
-            batch_size = 16
+            batch_size = 15
```

Comment:
What is this change trying to test? It is skipped in the CI anyways so likely not the best place to test

Reply:
`self.assertEqual(test_dataset.total_iterations, size // batch_size)` below will actually fail due to a partial batch at the end when running with batch_size = 16. A different change would have added one to the assertEqual.

Note that I changed the total_iterations in ArmoryDataGenerator:
```
self.total_iterations = size // batch_size + bool(size % batch_size)
```

Reply:
Gotcha, could you modify that all tests in the dataset_test file:
`self.assertEqual(test_dataset.total_iterations, size // batch_size + bool(size % batch_size)`

Would prefer it's explicit what we're trying to test instead of depending on the batch_size vs. total size ratio.

Reply:
Yes, I can do that.

Reply:
Only needed to do it one other place. The remaining ones are simple assertions like 
`self.assertEqual(test_dataset.total_iterations, 100)`, e.g., for MNIST.